x/delayedack/keeper/keeper.go

```diff
@@ -187,19 +187,26 @@ func (k *Keeper) LookupModuleByChannel(ctx sdk.Context, portID, channelID string
 }
 
 // ValidateRollappId checks that the rollappid from the ibc connection matches the rollapp checking the sequencer registered with the consensus state validator set
-func (k *Keeper) ValidateRollappId(ctx sdk.Context, rollapp, portID, channelID string) error {
+func (k *Keeper) ValidateRollappId(ctx sdk.Context, rollappID, portID, channelID string) error {
 	// Get the sequencer from the latest state info update and check the validator set hash
-	// from the headers match with the sequencer for the rollapp
+	// from the headers match with the sequencer for the rollappID
 	// As the assumption the sequencer is honest we don't check the packet proof height.
-	latestStateIndex, found := k.rollappKeeper.GetLatestStateInfoIndex(ctx, rollapp)
+	latestStateIndex, found := k.rollappKeeper.GetLatestStateInfoIndex(ctx, rollappID)
 	if !found {
-		return errorsmod.Wrapf(rollapptypes.ErrUnknownRollappID, "state index not found for the rollapp: %s", rollapp)
+		return errorsmod.Wrapf(rollapptypes.ErrUnknownRollappID, "state index not found for the rollappID: %s", rollappID)
 	}
-	stateInfo, found := k.rollappKeeper.GetStateInfo(ctx, rollapp, latestStateIndex.Index)
+	stateInfo, found := k.rollappKeeper.GetStateInfo(ctx, rollappID, latestStateIndex.Index)
 	if !found {
-		return errorsmod.Wrapf(rollapptypes.ErrUnknownRollappID, "state info not found for the rollapp: %s with index: %d", rollapp, latestStateIndex.Index)
+		return errorsmod.Wrapf(rollapptypes.ErrUnknownRollappID, "state info not found for the rollappID: %s with index: %d", rollappID, latestStateIndex.Index)
+	}
+	rollapp, found := k.rollappKeeper.GetRollapp(ctx, rollappID)
+	if !found {
+		return errorsmod.Wrapf(rollapptypes.ErrUnknownRollappID, "rollappID not found for the rollappID: %s", rollappID)
+	}
+	// check if the channelID matches the rollappID's channelID
+	if rollapp.ChannelId != channelID {
```

Comment:
It means that w/o the genesis event, no ibc transfer will work for this rollapp.
right?

Reply:
IMO this check should be done in `ExtractRollappIDAndTransferPacket`
* validates the rollappID before returning it
* no need for additional `k.rollappKeeper.GetRollapp` call

Reply:
Works for me